ccmlib/scylla_repository.py

```diff
@@ -234,15 +234,40 @@ def setup(version, verbose=True, skip_downloads=False):
               - release:4.2.1
               - release:2020.1
               - release:2020.1.5
+            flags can be added to the version using [(flag(=value)?)+]
+              - supported flags are:
+                - product=(scylla|scylla-enterprise)
+                - arch=(x86_64|aarch64)
+                - debug
 
     :param verbose: if True, print progress during download
     :param skip_downloads: if True, skips the actual download of files for testing purposes
 
     """
     s3_url = ''
-    type_n_version = version.split(':', 1)
-    scylla_product = os.environ.get('SCYLLA_PRODUCT', 'scylla')
-    scylla_arch = os.environ.get('SCYLLA_ARCH', 'x86_64')
+    version_re = re.compile(r"""
+                            (?P<type>
+                                \w+(?:/(?:[\w\d.~-]+))?
+                            )
+                            :(?P<version>[\w\d:.~-]+)
+                            (?:
+                                \[
+                                    (?P<flags>[^\]]*)
+                                \]
+                            )?""", re.VERBOSE)
+    m = version_re.match(version)
+    assert m, f"Invalid version {version}: use (release:version|unstable/branch:(latest|timestamp))([(flag(=value)?)*])? where flags can be any of: (product|arch|debug)"
+    type_n_version = [m.group("type"), m.group("version")]
+    flags = dict()
+    if m.group("flags"):
+        for kv in m.group("flags").split(","):
+            t = kv.split("=", 1)
+            key = t[0].strip().lower()
+            value = t[1].strip() if len(t) > 1 else True
+            flags[key] = value
+    scylla_product = flags.get("product", os.environ.get('SCYLLA_PRODUCT', 'scylla'))
+    scylla_arch = flags.get("arch", os.environ.get('SCYLLA_ARCH', 'x86_64'))
```

Comment:
```suggestion
    scylla_arch = flags.get("arch", os.environ.get('SCYLLA_ARCH', platform.uname().machine))
```
A better default that includes those that might have arm based systems.

Reply:
Makes sense.  It's somewhat out of scope for this PR, but  an be incorporated into the series whilr making changes in this area.

Reply:
In dtest we do that

So we can pick the right docker image:

https://github.com/scylladb/scylla-dtest/blob/f58d7b89534ad895ee7d1ec54c404414eedcceca/scripts/run_test.sh#L108

Reply:
> In dtest we do that
> 
> So we can pick the right docker image:
> 
> [scylladb/scylla-dtest@`f58d7b8`/scripts/run_test.sh#L108](https://github.com/scylladb/scylla-dtest/blob/f58d7b89534ad895ee7d1ec54c404414eedcceca/scripts/run_test.sh#L108)

meaning?
This seems to support @cezarmoise's suggestion, right?

Reply:
Added in v3